### PR TITLE
Fix (log)pdf of VonMises

### PR DESCRIPTION
**Pull request by bdeonovic**
Added the correct support check. Fixes https://github.com/JuliaStats/Distributions.jl/issues/1022

### CODE REVIEW

**Comment by matbesancon** [comment]
Thanks for the fix, could you add a test, for example with a shifted von Mises, `µ = 10`, that the pdf is > 0 at the two boundaries

**Comment by bdeonovic** [comment]
where is the most appropriate place to put it?

**Comment by matbesancon** [comment]
You can create a vonmises.jl test file, I don't think there is already somewhere appropriate

**Comment by matbesancon** [comment]
ping @bdeonovic :)

**Comment by matbesancon** [comment]
bump

**Comment by bdeonovic** [comment]
heyo! I will finish up this PR this week!

**Comment by matbesancon** [comment]
Great thanks :)

File: test/vonmises.jl [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Tests for Von-Mises distribution
+
+using Distributions
+using Compat.Test
```
Comment (matbesancon) [inline test/vonmises.jl]:
```suggestion
using Test
```
Do we need legacy Test behavior here?

Comment (matbesancon) [reply test/vonmises.jl]: @bdeonovic this is the last thing before merging I think

**Comment by bdeonovic** [comment]
No I just copy posted from another test file. I'll update after lunch.
Thanks

**Comment by matbesancon** [comment]
Thanks!
